### PR TITLE
Updating the updating of vrm item as it was not adding the created_at…

### PR DESCRIPTION
**Pull request by Thomas-Franklin**
… date, or updated_at column and additional unit tests

### CODE REVIEW

File: src/main/java/uk/gov/dvsa/motr/notifier/component/subscription/persistence/DynamoDbSubscriptionRepository.java [hunk 1]
```diff
@@ -101,14 +101,16 @@ public void updateVrm(String vrm, String email, String updatedVrm) {
...
                     .withString("mot_test_number", originalItem.getString("mot_test_number"))
                     .withString("mot_due_date", originalItem.getString("mot_due_date"))
-                    .withString("mot_due_date_md", originalItem.getString("mot_due_date_md"));
+                    .withString("mot_due_date_md", originalItem.getString("mot_due_date_md"))
+                    .withString("created_at", originalItem.getString("created_at"))
+                    .withString("updated_at", ZonedDateTime.now().format(DateTimeFormatter.ISO_INSTANT));
             dynamoDb.getTable(tableName).putItem(item);
         } catch (Exception e) {
-            logger.error("UPDATING-VRM-ERROR: {} => {}, {}", vrm, updatedVrm, e.getMessage());
+            throw new Exception(e);
```
Comment (aaronlong) [inline src/main/java/uk/gov/dvsa/motr/notifier/component/subscription/persistence/DynamoDbSubscriptionRepository.java]: More a question here:  If this tests and throws an exception will re-throwing the exception here stop the rest of the tests from running?  As the exception was previously being suppressed by the catch.

File: src/integration-test/java/uk/gov/dvsa/motr/test/integration/unloader/SubscriptionDbItemQueueItemUnloaderTest.java [hunk 2]
```diff
@@ -117,19 +120,24 @@ public void whenProcessingASubscriptionWithAMismatchedVrm_TheSubscriptionVrmInTh
...
+
+        Item savedItem = new DynamoDB(dynamoDbClient()).getTable(subscriptionTableName()).query(spec).iterator().next();
+
         // Assert the new  db item has the same id subscriptionItem
         // (vrm update requires new record to be created, but want to keep original id).
         assertEquals(subscriptionItem.getId(), changedSubscriptionDbItem.getId());
 
         // Assert the db vrm now is equal to the mock api vrm.
-        assertTrue(changedSubscriptionDbItem.getVrm().equals("WDD2040022A65"));
+        assertTrue(changedSubscriptionDbItem.getVrm().equals("XXXYYY"));
```
Comment (mathieuhendey) [inline src/integration-test/java/uk/gov/dvsa/motr/test/integration/unloader/SubscriptionDbItemQueueItemUnloaderTest.java]: Why not assertEquals?

Comment (Thomas-Franklin) [inline src/integration-test/java/uk/gov/dvsa/motr/test/integration/unloader/SubscriptionDbItemQueueItemUnloaderTest.java]: Yea I personally prefer assertEquals, I'll update now

File: src/test/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionDbItemQueueItemServiceTest.java [hunk 3]
```diff
@@ -30,7 +30,9 @@
     private String webBaseUrl = "http://gov.uk";
 
     private static final String TEST_VRM = "TEST-VRM";
+    private static final String UPDATED_VRM = "NEW-REG";
```
Comment (mathieuhendey) [inline src/test/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionDbItemQueueItemServiceTest.java]: const name doesn't seem to match value?

File: src/test/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionDbItemQueueItemServiceTest.java [hunk 4]
```diff
@@ -148,12 +150,88 @@ public void whenStoredExpiryDateIsAfterTheVehicleApiButWithinOneMonthReminder_Th
...
     }
 
+    /**
+     * When MOTR VRM is different from Trade API then the record is updated
+     * @throws Exception
+     */
+    @Test
+    public void whenStoredVrmIsDifferentFromTheVehicleApi_ThenRecordIsUpdated() throws Exception {
+
+        LocalDate vehicleExpiryDate = LocalDate.of(2017, 10, 24);
```
Comment (mathieuhendey) [inline src/test/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionDbItemQueueItemServiceTest.java]: You seem to be creating one of these in every test, why not create it in a setup method?

Comment (Thomas-Franklin) [inline src/test/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionDbItemQueueItemServiceTest.java]: So the date changes as the tests are testing different things, so we kept the control within the test. In saying so we could split them out in to consts but I feel for the flow of the test it is best to keep them controlled within.

Comment (mathieuhendey) [inline src/test/java/uk/gov/dvsa/motr/notifier/processing/service/ProcessSubscriptionDbItemQueueItemServiceTest.java]: (y)

**Pull request merged**